### PR TITLE
Check the existence of home dir reported by env vars on Windows

### PR DESCRIPTION
**Pull request by fabianofranz**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1305765

Sometimes the home directory env vars on Windows can point to directories that actually don't exist. We need to check the existence of what's reported as home dir when on the Windows platform, and fallback to `$HOME` if nothing else works.

### CODE REVIEW

**Comment by fabianofranz** [comment]
[merge]

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/homedir/homedir.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"os"
+	"runtime"
+)
+
+// HomeDir returns the home directory for the current user
+func HomeDir() string {
+	if runtime.GOOS == "windows" {
+		if homeDrive, homePath := os.Getenv("HOMEDRIVE"), os.Getenv("HOMEPATH"); len(homeDrive) > 0 && len(homePath) > 0 {
+			homeDir := homeDrive + homePath
```
Comment (liggitt) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/homedir/homedir.go]: filepath.join?

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/unversioned/clientcmd/loader.go [hunk 2]
```diff
@@ -44,8 +45,8 @@ const (
 	RecommendedSchemaName       = "schema"
 )
 
-var RecommendedHomeFile = path.Join(HomeDir(), RecommendedHomeDir, RecommendedFileName)
-var RecommendedSchemaFile = path.Join(HomeDir(), RecommendedHomeDir, RecommendedSchemaName)
+var RecommendedHomeFile = path.Join(homedir.HomeDir(), RecommendedHomeDir, RecommendedFileName)
+var RecommendedSchemaFile = path.Join(homedir.HomeDir(), RecommendedHomeDir, RecommendedSchemaName)
```
Comment (liggitt) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/unversioned/clientcmd/loader.go]: Should these use filepath.Join?

Comment (fabianofranz) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/unversioned/clientcmd/loader.go]: Hrm, makes sense, but I'll have to test. Those are here for quite some time and we didn't have reports of issues, but let me do some testing on Windows as a follow-up of this - if it works, it seems to make better sense than just using `path.Join`.

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/homedir/homedir.go (hunk 1 above)
Comment (fabianofranz) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/homedir/homedir.go]: The recommendation when building a homedir based on these two env vars is just to concatenate<i><sup>citation needed</sup></i>, so I think no.

**Pull request merged**